homeassistant/components/knx/expose.py

```diff
@@ -132,6 +139,9 @@ def _get_expose_value(self, state: State | None) -> StateType:
             and issubclass(self.device.sensor_value.dpt_class, DPTNumeric)
         ):
             return float(value)
+        if self.type == "string":
```

Comment:
We should check for `isinstance(self.device.sensor_value.dpt_class, DPTString)` like above for numeric to also catch eg. `self.type = "16.000"`.
And `value is not None` to avoid sending `"None"`.